Prog/2023/Sales_affordability_2016_22.sas

```diff
@@ -0,0 +1,415 @@
+/**************************************************************************
+ Program:  Sales_Affordability_2016_22.sas
+ Library:  Equity
+ Project:  NeighborhoodInfo DC
+ Author:   M. Woluchem	
+ Created:  8/12/16
+ Version:  SAS 9.2
+ Environment:  Windows with SAS/Connect
+ 
+ Description: *Methodology for affordability adapted from Zhong Yi Tong paper 
+http://content.knowledgeplex.org/kp2/cache/documents/22736.pdf
+Homeownership Affordability in Urban America: Past and Future;
+
+ Modifications: 09/11/16 LH Added Price Adjustments and used 2015$ adj. income
+				10/07/16 LH Added output for COMM. 
+				03/18/22 LH Update for 2016-2020
+				09/13/22 RP Update for 2020 tracts and 2022 wards
+				05/12/23 RP Update for most recent real property data
+
+**************************************************************************/
+
+
+%include "\\sas1\DCdata\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( Requests )
+%DCData_lib( realprop );
+%DCData_lib( equity );
+
+%let firstyear = 2016;
+%let lastyear = 2022;
+
+%let yrs = %sysfunc(substr(&firstyear, 1, 4))_%sysfunc(substr(&lastyear, 3, 2));
+%let yr_label = %sysfunc(substr(&firstyear, 1, 4))-%sysfunc(substr(&lastyear, 3, 2));
+
+
+data create_flags;
+	set realpr_r.sales_res_clean (where=(ui_proptype in ('10' '11') and &firstyear. <= year(saledate) <= &lastyear.))
+	;
+
+	sale_yr = year(saledate);
+  
+  /* Copy/paste in effective interest rates by downloading the Current Mortgage Rates Data Since 1971? (xlsx) from:
+     https://www.freddiemac.com/pmms */;
+  
+	eff_int_rate_2016= 3.72;
+	eff_int_rate_2017= 4.06;
+	eff_int_rate_2018= 4.64;
+	eff_int_rate_2019= 4.01;
+	eff_int_rate_2020= 3.11;
```

Comment:
These should be the effective vs. nominal rates - update for 2020-2022. https://urbanorg.app.box.com/file/933065867963